app/src/main/java/com/asfoundation/wallet/ui/iab/BillingWebViewFragment.kt

```diff
@@ -0,0 +1,179 @@
+package com.asfoundation.wallet.ui.iab
+
+import android.content.ActivityNotFoundException
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.CookieManager
+import android.webkit.WebResourceRequest
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import com.asf.wallet.R
+import com.asfoundation.wallet.billing.analytics.BillingAnalytics
+import com.google.android.material.snackbar.Snackbar
+import dagger.android.support.DaggerFragment
+import kotlinx.android.synthetic.main.webview_fragment.*
+import kotlinx.android.synthetic.main.webview_fragment.view.*
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.atomic.AtomicReference
+import javax.inject.Inject
+
+class BillingWebViewFragment : DaggerFragment() {
+  private val timeoutReference: AtomicReference<ScheduledFuture<*>?> = AtomicReference()
+  @Inject
+  lateinit var inAppPurchaseInteractor: InAppPurchaseInteractor
+  @Inject
+  lateinit var analytics: BillingAnalytics
+  private var currentUrl: String? = null
+  private var executorService: ScheduledExecutorService? = null
+  private var webViewActivity: WebViewActivity? = null
+  private var asyncDetailsShown = false
+
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    require(context is WebViewActivity) { "WebView fragment must be attached to WebView Activity" }
+    webViewActivity = context
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    executorService = Executors.newScheduledThreadPool(0)
+    require(!(arguments == null || !arguments!!.containsKey(
+        URL))) { "Provided url is null!" }
```

Comment:
- [ ] Code style

- [ ] Considering !(A = B || !C) => (A != B && C) we should change it since it improves readability

Reply:
Done